src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStoreV1.swift

```diff
@@ -165,6 +168,28 @@ class ContactDiaryStoreV1: DiaryStoring, DiaryProviding {
 		return result
 	}
 
+	@discardableResult
+	func cleanup(timeout: TimeInterval) -> DiaryStoringVoidResult {
+		let group = DispatchGroup()
+		var result: DiaryStoringVoidResult?
+
+		group.enter()
+		DispatchQueue.global().async {
+			result = self.cleanup()
+			group.leave()
+		}
+
+		guard group.wait(timeout: DispatchTime.now() + timeout) == .success else {
+			databaseQueue.interrupt()
+			return .failure(.unknown)
```

Comment:
Would 
```suggestion
			return .failure(.timeout)
```
Be possible?

Reply:
Added timeout error type.